sys/kern/subr_sleepqueue.c

```diff
@@ -1457,11 +1457,11 @@ DB_SHOW_COMMAND(sleepq, db_show_sleepqueue)
 	 */
 	for (i = 0; i < SC_TABLESIZE; i++)
 		LIST_FOREACH(sq, &sleepq_chains[i].sc_queues, sq_hash) {
-			if (sq == (struct sleepqueue *)addr)
+			if ((ptraddr_t)sq == addr)
 				goto found;
 		}
 
-	db_printf("Unable to locate a sleep queue via %p\n", (void *)addr);
+	db_printf("Unable to locate a sleep queue via %#lx\n", (long)addr);
```

Comment:
I'd prefer:
```suggestion
	db_printf("Unable to locate a sleep queue via %p\n", (void *)(uintptr_t)addr);
```
or (since it's the same thing, shorter, and the thing we actually use when searching):
```suggestion
	db_printf("Unable to locate a sleep queue via %p\n", wchan);
```